### PR TITLE
Frontend/feat interoperability

### PR DESCRIPTION
**Pull request by spravinkumar9952**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/package.json
Comment (kavya-shree-s) [inline Frontend/ui-customer/package.json:1]: @spravinkumar9952 Can you undo these changes

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 1]
```diff
@@ -165,7 +171,7 @@ shareRideButton push state =
   [ height $ WRAP_CONTENT
   , width $ WRAP_CONTENT
   , orientation VERTICAL
-  , clickable true
+  , clickable $ state.data.providerType == ONUS -- true
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:174]: Remove comment

File: Frontend/ui-customer/src/Components/DriverInfoCard/View.purs [hunk 2]
```diff
@@ -175,7 +181,7 @@ shareRideButton push state =
      , background Color.white900
      , stroke $ "1,"<> Color.grey900
      , cornerRadius if os == "IOS" then 20.0 else 32.0
-     , clickable true
+     , clickable $ state.data.providerType == ONUS -- true
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Components/DriverInfoCard/View.purs:184]: here as well

File: Frontend/ui-customer/src/Resources/Localizable/HI.purs [hunk 3]
```diff
@@ -1111,3 +1111,22 @@ getHI stringKey =
...
+        SELECT_A_PROVIDER -> "एक प्रदाता चुनें"
+        CONFIRMING_SELECTED_PROVIDER -> "चयनित प्रदाता की पुष्टि की जा रही है:"
+        BOOK_TOP_PROVIDER -> "बुक टॉप प्रदाता"
+        CHOOSE_FROM_PROVIDERS -> "सभी उपलब्ध प्रदाताओं में से चुनें"
+        CHOOSE_BETWEEN_PROVIDERS -> "एकाधिक प्रदाताओं में से चुनें"
+        CHOOSE_BETWEEN_PROVIDERS_DESC -> "सभी उपलब्ध प्रदाताओं के उद्धरण देखने के लिए इस सुविधा को सक्षम करें।"
+        GUARANTEED_RIDE -> "गारंटीयुक्त सवारी"
+        RIDE_FULFILLED_BY provider -> "यह सवारी पूरी होती है:" <> provider
+        ADDITIONAL_FEATURES_ON merchantName -> merchantName <> " पर अतिरिक्त सुविधाएं"
+        NOTIFY_YOUR_EC -> ""
```
Comment (kavya-shree-s) [inline Frontend/ui-customer/src/Resources/Localizable/HI.purs:1129]: Can you update translation

File: Frontend/ui-customer/src/Components/ChooseYourRide/View.purs [hunk 4]
```diff
@@ -543,26 +564,39 @@ estimatedTimeAndDistanceView push config =
...
+                [ height WRAP_CONTENT
+                , width MATCH_PARENT
+                , orientation VERTICAL
+                ] $ map( \item -> ChooseVehicle.view (push <<< ChooseVehicleAC)  item {singleVehicle = (length variantBasedList == 1)}) (spy "variantBasedList" (  variantBasedList))
+              else 
+                Tuple "TopProvider" $ linearLayout
+                [ height WRAP_CONTENT
+                , width MATCH_PARENT
+                , orientation VERTICAL
+                ] $ map (\item -> ChooseVehicle.view (push <<< ChooseVehicleAC) item{showInfo = true, singleVehicle = (length topProviderList == 1)}) (spy "topProviderList" topProviderList)
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Components/ChooseYourRide/View.purs:597]: Remove spy

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 5]
```diff
@@ -1860,6 +1952,11 @@ eval OpenEmergencyHelp state = do
   let _ = unsafePerformEffect $ logEvent state.data.logField "ny_ic_safety_center_clicked"
   exit $ GoToNammaSafety state true false
 
+eval OpenOffUsSOS state = do
+  _ <- pure $ performHapticFeedback unit
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:1956]: Use void instead of `_` 
In other files as well

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 6]
```diff
@@ -2364,8 +2468,9 @@ eval (ContinueWithoutOffers (SelectListRes resp)) state = do
           exit $ ConfirmRide state{props{currentStage = ConfirmingRide, bookingId = bookingId, isPopUp = NoPopUp, selectedQuote = Nothing}}
     Nothing -> do
       if isLocalStageOn QuoteList then do
-        let updatedState = if isTipEnabled state then tipEnabledState state{props{isPopUp = TipsPopUp, customerTip{enableTips = true}}} else state{props{isPopUp = ConfirmBack}}
-        continue updatedState
+        let onUs = state.data.selectedEstimatesObject.providerType == CTP.ONUS
+            updatedState = if (isTipEnabled state && onUs) then tipEnabledState state{props{isPopUp = TipsPopUp, customerTip{enableTips = true}}} else state{props{isPopUp = ConfirmBack}} -- TODO // check in what case we need to show tip after seearch expired
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:2472]: Can you confirm this case as well ?

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 7]
```diff
@@ -3086,43 +3228,90 @@ specialZoneFlow estimatedQuotes state = do
...
-      estimatesInfo = getEstimatesInfo estimates "" newState
-  if ((not (null estimatesInfo.quoteList)) && (isLocalStageOn FindingEstimate)) then do
-    let _ = unsafePerformEffect $ logEvent state.data.logField "ny_user_quote"
-        nearByDrivers = getNearByDrivers estimates
-        nearByDriversLength = length nearByDrivers
-    _ <- pure $ updateLocalStage SettingPrice
+estimatesListFlow :: Array EstimateAPIEntity -> HomeScreenState -> Int -> Eval Action ScreenOutput HomeScreenState
+estimatesListFlow estimates state count = do
+  let 
+    repeatRideFailCheck = isJust state.props.repeatRideServiceTierName && not checkRecentRideVariantInEstimates estimates state.props.repeatRideServiceTierName -- check if the repeat ride variant is available in the estimates
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:3234]: Please check these cases And
Remove comments if resolved (Everywhere in this file)

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Transformer.purs [hunk 8]
```diff
@@ -60,6 +60,8 @@ import Data.Either (Either(..))
 import Screens.NammaSafetyFlow.Components.SafetyUtils (getDefaultPriorityList)
 import Mobility.Prelude as MP
 import Data.Function.Uncurried (runFn2)
+import Common.Types.App as CT
+import Debug
```
Comment (Rohit4997) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Transformer.purs:64]: Remove debug import

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs (hunk 7 above)
Comment (spravinkumar9952) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:3234]: added comments for Readability

**Comment by kavya-shree-s** [comment]
@spravinkumar9952 Can you resolve conflicts

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs (hunk 7 above)
Comment (Rohit4997) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:3234]: Not talking about this
We so many comments as TODO in the PR 
please check those

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 9]
```diff
@@ -160,6 +163,7 @@ screen initialState =
             else do
               pure unit
             when (isNothing initialState.data.bannerData.bannerItem) $ void $ launchAff $ flowRunner defaultGlobalState $ computeListItem push
+            void $ pure $ spy "CURRENT_STAGE" initialState.props.currentStage
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:166]: Remove logs

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Transformer.purs [hunk 10]
```diff
@@ -450,15 +463,43 @@ getEstimates (EstimateAPIEntity estimate) index isFareRange =
...
       , additionalFare = additionalFare
       , nightShiftMultiplier = nightShiftMultiplier
       , nightCharges = nightCharges
       , baseFare = baseFare
+      , providerName = fromMaybe "" estimate.providerName
+      , providerId = fromMaybe "" estimate.providerId
+      , providerType = maybe CT.OFFUS (\valueAdd -> if valueAdd then CT.ONUS else CT.OFFUS) estimate.isValueAddNP
+      , maxPrice = extractFare _.maxFare
+      , minPrice = extractFare _.minFare
+      , priceShimmer = if count == 1 then false else true
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Transformer.purs:478]: Assign `count == 1` directly

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 11]
```diff
@@ -3086,43 +3228,90 @@ specialZoneFlow estimatedQuotes state = do
...
+    exit $ SelectEstimate newState 
       { data
-        { specialZoneQuoteList = quoteList
+        { specialZoneQuoteList = quoteList'
         , currentSearchResultType = ESTIMATES
-        , selectedEstimatesObject = estimatesInfo.defaultQuote
+        , selectedEstimatesObject = defaultQuote
         , nearByDrivers = if nearByDriversLength > 0 then Just nearByDriversLength else Nothing
+        , iopState { 
+          showPrefButton = state.data.currentCityConfig.iopConfig.enable && (not (null topProviderEstimates)) && (not newState.props.isRepeatRide)
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:3291]: Put `state.data.currentCityConfig.iopConfig.enable && (not (null topProviderEstimates)) && (not newState.props.isRepeatRide)` in let

File: Frontend/ui-customer/src/Screens/NammaSafetyFlow/SafetySettingsScreen/View.purs [hunk 12]
```diff
@@ -413,3 +575,52 @@ toggleSwitchView isActive visibility' action push =
...
+    , visibility $ boolToVisibility state.props.showShimmer
+    ]
+    [ sfl (V 400) 16 1 true
+    , linearLayout
+        [ width MATCH_PARENT
+        , height WRAP_CONTENT
+        , orientation VERTICAL
+        , alignParentBottom "true,-1"
+        ]
+        [ sfl (V 80) 130 3 (getValueToLocalStore IS_SOS_ACTIVE == "true")
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Screens/NammaSafetyFlow/SafetySettingsScreen/View.purs:595]: Put this condition in let and then use in both the places

File: Frontend/ui-customer/src/Screens/NammaSafetyFlow/SafetySettingsScreen/View.purs [hunk 13]
```diff
@@ -131,34 +133,193 @@ view push state =
...
+            , visibility GONE
+            ]
+            [ imageView
+                [ imageWithFallback $ fetchImage FF_ASSET "ic_namma_yatri_logo"
+                , height $ V 20
+                , width $ V 20
+                , margin $ MarginRight 8
+                ]
+            , textView
+                $ [ text "Guaranteed Ride"
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Screens/NammaSafetyFlow/SafetySettingsScreen/View.purs:271]: Add translations

File: Frontend/ui-customer/src/Screens/NammaSafetyFlow/SafetySettingsScreen/Controller.purs [hunk 14]
```diff
@@ -206,4 +208,10 @@ eval (CheckRideListResp (RideBookingListRes listResp)) state = do
           mbRideData = DA.head transformedResp
       continue state{props{showPastRidePopUp = isRecentRide}, data{lastRideDetails = mbRideData}}
  
+eval _ state = continue state
+eval DialPolice state = do
+   void $ pure $ HU.performHapticFeedback unit
+   pure $ JB.showDialer "112" false
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Screens/NammaSafetyFlow/SafetySettingsScreen/Controller.purs:214]: Can we put this "112" in constants file

File: Frontend/ui-customer/src/Screens/CustomerUtils/RideSelectionScreen/Transformer.purs [hunk 15]
```diff
@@ -90,7 +91,7 @@ myRideListTransformer isSrcServiceable listRes = filter (\item -> (item.status =
     city = getCityFromString $ getValueToLocalStore CUSTOMER_LOCATION
     nightChargeFrom = if city == Delhi then "11 PM" else "10 PM"
     nightChargeTill = "5 AM"
-    referenceString' = (if nightChargesVal && (getMerchant FunctionCall) /= YATRI then "1.5" <> (getEN $ DAYTIME_CHARGES_APPLICABLE_AT_NIGHT nightChargeFrom nightChargeTill) else "")
+    referenceString' = (if nightChargesVal && (getMerchant CTP.FunctionCall) /= YATRI then "1.5" <> (getEN $ DAYTIME_CHARGES_APPLICABLE_AT_NIGHT nightChargeFrom nightChargeTill) else "")
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Screens/CustomerUtils/RideSelectionScreen/Transformer.purs:94]: Why is this getEN?

File: Frontend/ui-customer/src/Components/QuoteListModel/View.purs [hunk 16]
```diff
@@ -735,4 +749,91 @@ tipsViewConfig state = let
...
+      ]
+
+  ]
+
+providerModelConfig :: QuoteListModelState -> CVC.Config -> PM.Config
+providerModelConfig config quoteItem = 
+  let onus = quoteItem.providerType == ONUS
+  in PM.config {
+    isActive = quoteItem.id == config.selectedEstimatesObject.id,
+    pillsVisibility = if onus then VISIBLE else GONE,
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Components/QuoteListModel/View.purs:829]: use `boolToVisibility`

File: Frontend/ui-customer/src/Components/QuoteListModel/View.purs [hunk 17]
```diff
@@ -735,4 +749,91 @@ tipsViewConfig state = let
...
+      [ width MATCH_PARENT
+      , height WRAP_CONTENT
+      , orientation VERTICAL
+      ][  animationtimer push state
+        , linearLayout
+          [ width MATCH_PARENT
+          , height WRAP_CONTENT
+          , margin $ Margin 16 16 16 16
+          , orientation VERTICAL
+          ](map (\element -> PM.view (push <<< ProviderModelAC) $ spy "providerModelConfig" (providerModelConfig state element) ) $  spy "filteredWithSelectedVariant" filteredWithSelectedVariant)
```
Comment (aryan-13) [inline Frontend/ui-customer/src/Components/QuoteListModel/View.purs:770]: Remove logs

**Review by aryan-13 — CHANGES_REQUESTED** [review]

**Review by kavya-shree-s — APPROVED** [review]
